yt_dlp/extractor/rinsefm.py

```diff
@@ -0,0 +1,45 @@
+import re
+import json
+from datetime import datetime, timezone
+
+from .common import InfoExtractor
+
+
+class RinseFMIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?rinse\.fm/episodes/(?P<id>.+)'
+    _TESTS = [{
+        'url': 'https://rinse.fm/episodes/cameo-blush-01-09-2023-2300/',
+        'md5': '9284abbd785e6b86e67d1cdca6224feb',
+        'info_dict': {
+            'id': '1351562',
+            'ext': 'mp3',
+            'title': 'Cameo Blush - 01/09/2023 - 23:00',
+            'thumbnail': r're:^https?://.*\.JPG$',
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        response = re.findall(r'<script[^>]*id\s*=\s*[\'"]__NEXT_DATA__[\'"][^>]*>(.*?)<\/script>', webpage, re.DOTALL)
+        info = json.loads(response[0])
```

Comment:
```suggestion
        info = self._search_nextjs_data(webpage, video_id)
```

Reply:
Thank you, I didn't realize there was a built-in for nextjs.
I used it and all tests pass.